src/client/oicp/CpoOICPClient.ts

```diff
@@ -0,0 +1,1272 @@
+import { OICPChargingNotificationEndCpoSend, OICPChargingNotificationErrorCpoSend, OICPChargingNotificationProgressCpoSend, OICPChargingNotificationStartCpoSend } from '../../types/oicp/OICPChargingNotifications';
+import { OICPEVSEPricing, OICPPricingProductData } from '../../types/oicp/OICPPricing';
+import { OICPEvseDataRecord, OICPEvseStatusRecord, OICPOperatorEvseData, OICPOperatorEvseStatus } from '../../types/oicp/OICPEvse';
+import { OICPIdentification, OICPRFIDmifarefamilyIdentification } from '../../types/oicp/OICPIdentification';
+import { OICPPushEVSEPricingCpoSend, OICPPushPricingProductDataCpoSend } from '../../types/oicp/OICPDynamicPricing';
+import { OICPSession, OICPSessionStatus } from '../../types/oicp/OICPSession';
+import Transaction, { TransactionAction } from '../../types/Transaction';
+
+import BackendError from '../../exception/BackendError';
+import Constants from '../../utils/Constants';
+import { HTTPError } from '../../types/HTTPError';
+import Logging from '../../utils/Logging';
+import OCPPStorage from '../../storage/mongodb/OCPPStorage';
+import { OICPAcknowledgment } from '../../types/oicp/OICPAcknowledgment';
+import { OICPAuthorizationStatus } from '../../types/oicp/OICPAuthentication';
+import { OICPChargeDetailRecord } from '../../types/oicp/OICPChargeDetailRecord';
+import OICPClient from './OICPClient';
+import OICPEndpoint from '../../types/oicp/OICPEndpoint';
+import OICPEndpointStorage from '../../storage/mongodb/OICPEndpointStorage';
+import { OICPJobResult } from '../../types/oicp/OICPJobResult';
+import OICPMapping from '../../server/oicp/oicp-services-impl/oicp-2.3.0/OICPMapping';
+import { OICPPushEvseStatusCpoSend } from '../../types/oicp/OICPEvseStatus';
+import { OICPRole } from '../../types/oicp/OICPRole';
+import { OicpSetting } from '../../types/Setting';
+import { ServerAction } from '../../types/Server';
+import SiteAreaStorage from '../../storage/mongodb/SiteAreaStorage';
+import Tenant from '../../types/Tenant';
+import _ from 'lodash';
+
+const MODULE_NAME = 'CpoOICPClient';
+
+export default class CpoOICPClient extends OICPClient {
+
+  constructor(tenant: Tenant, settings: OicpSetting, oicpEndpoint: OICPEndpoint) {
+    super(tenant, settings, oicpEndpoint, OICPRole.CPO);
+    if (oicpEndpoint.role !== OICPRole.CPO) {
+      throw new BackendError({
+        message: `CpoOicpClient requires Oicp Endpoint with role ${OICPRole.CPO}`,
+        module: MODULE_NAME, method: 'constructor',
+      });
+    }
+  }
+
+  async startSession(chargingStation: ChargingStation, transaction: Transaction): Promise<void> {
+    // Build payload
+    // Tbd: Hubject differentiates between different identification Methods. Eg. RFID, Remote start, QR Code, Plug&Charge
+    const identification: OICPIdentification = {} as OICPIdentification;
+    const rfidIdentification: OICPRFIDmifarefamilyIdentification = {} as OICPRFIDmifarefamilyIdentification;
```

Comment:
Maybe you can explain to me what you would do exactly when you have time

Reply:
At some point, you will have to define the list of string or number in OICP spec for: 
+ OICPIdentification
+ OICPRFIDmifarefamilyIdentification
if not done alreay.

Since you are not allowing yet to define them in startSession() as optional arguments, you must define a fallback to sensible default values for both. Sensible means the most used at the time: RFID with the most used family id.

Reply:
I spoke to Hubject:
Since we get no information about the card type (mifareCls, mifareDes, calypso, nfc, mifareFamily), we can always fallback on mifareFamily